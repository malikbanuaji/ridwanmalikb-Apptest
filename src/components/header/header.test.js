import * as React from 'react';
import Header from './header';

import {render} from '@testing-library/react-native';

test('should render properly', () => {
  const component = render(<Header title="Contact" />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render with right element', () => {
  const RightElement = 'RightElement';
  const component = render(
    <Header canGoBack title="Contact" rightElement={() => <RightElement />} />,
  );
  expect(component.toJSON()).toMatchSnapshot();
});
