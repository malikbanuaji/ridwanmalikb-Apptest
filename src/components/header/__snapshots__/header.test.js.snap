// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#36393f",
      "flexDirection": "row",
      "minHeight": 80,
      "paddingHorizontal": 10,
      "paddingVertical": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 22,
          "fontWeight": "700",
          "letterSpacing": 0.4,
        }
      }
    >
      Contact
    </Text>
  </View>
  <View />
</View>
`;

exports[`should render with right element 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#36393f",
      "flexDirection": "row",
      "minHeight": 80,
      "paddingHorizontal": 10,
      "paddingVertical": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <AntDesignIcon
      color="#ffffff"
      name="left"
      onPress={[Function]}
      size={20}
      style={
        Object {
          "padding": 15,
        }
      }
      testID="HeaderIcon"
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 0,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 22,
          "fontWeight": "700",
          "letterSpacing": 0.4,
        }
      }
    >
      Contact
    </Text>
  </View>
  <View>
    <RightElement />
  </View>
</View>
`;
